### PR TITLE
Added lighting effects to the welder.

### PR DESCRIPTION
**Pull request by TheCarlSaganExpress**
All welders now officially emit a light source when turned on.

### CODE REVIEW

**Comment by GunHog** [comment]
https://github.com/tgstation/tgstation/pull/19614

**Comment by TheCarlSaganExpress** [comment]
I actually came across the exact same issues and strangely enough originally did a very similar implementation two days ago. But, there were issues with the code. In #19614 there exists a glitch where if the welder runs out of fuel while in the players hand, the player will be completely illuminated for the rest of the round until the player toggles the welder again. There are also other issues, such as if the welder runs out of fuel it will still retain it's illumination effect as if it were turned on, despite being out of fuel. I actually came across all those issues and fixed them.

File: code/game/objects/items/weapons/tools.dm [hunk 1]
```diff
@@ -317,17 +317,26 @@
...
 			if(L.IgniteMob())
 				message_admins("[key_name_admin(user)] set [key_name_admin(L)] on fire")
 				log_game("[key_name(user)] set [key_name(L)] on fire")
 
+
 /obj/item/weapon/weldingtool/attack_self(mob/user)
-	toggle(user)
+	switched_on(user)
+	if(welding)
+		SetLuminosity(0)
```
Comment (optimumtact) [inline code/game/objects/items/weapons/tools.dm]: do these need to still be here if switched on is called above?

Comment (TheCarlSaganExpress) [inline code/game/objects/items/weapons/tools.dm]: Yeah, they still need to be there. Without those lines of code if you picked up a welder that was already turned on it wouldn't illuminate the player.

**Comment by optimumtact** [comment]
This is better, but I might cast my eye over it later and do a refactor.

Thanks for your work.

**Pull request merged**